src/main/java/com/fasterxml/jackson/databind/deser/std/MapDeserializer.java

```diff
@@ -613,7 +613,19 @@ protected final void _readAndBindStringKeyMap(JsonParser p, DeserializationConte
                 if (useObjectId) {
                     referringAccumulator.put(key, value);
                 } else {
-                    result.put(key, value);
```

Comment:
Instead of checking for existence first, it's more efficient to use return value of `result.put()` (that is the old Object, if any) and check if it is not `null`.